src/Mvc/Mvc.TagHelpers/src/TextAreaTagHelper.cs

```diff
@@ -103,9 +103,9 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 if (tagBuilder.HasInnerHtml)
                 {
                     // Overwrite current Content to ensure expression result round-trips correctly.
-                    output.Content.SetHtmlContent(tagBuilder.InnerHtml);
+                    output.Content.SetHtmlContent(output.GetChildContentAsync().Result.GetContent());
```

Comment:
* This needs a test
* Change this TagHelper to override `ProcessAsync` instead and avoid the blocking call.

Reply:
Hi @pranavkm, I used `async` overload of Process instead of the regular one and did the required changes. 
Am finding myself hard for writing test to validate the behavior, and also some of the existing tests are failing due to this change, I tried to find the cause but am failing. 
Can you assist me in how should we validate this behaviour? 
Btw I have tested `TextAreaTagHelper` by using it as a tag directly in `cshtml` and it's working as expected. 

Thanks!

Reply:
@MCCshreyas my apologies I missed your code comment. Unfortunately I'm not going to be able to spend time debugging this (since it's fairly low priority for us). If you're able to continue working on this to completion, we can review and merge it. But having failing tests is almost never a great sign.

Reply:
Hi @pranavkm, I will fix the test as well, these days am quite busy. I will try to get this done in the upcoming weekend. I hope that is ok. 

Thanks!